components/pages/workPageLayout/WorkPageLayout.tsx

```diff
@@ -1,58 +1,59 @@
 "use client"
 
 import { useQuery } from "@tanstack/react-query"
-import React, { useState } from "react"
+import React, { useEffect } from "react"
 
-import { Button } from "@/components/shared/button/Button"
-import Player from "@/components/shared/publizonPlayer/PublizonPlayer"
-import ResponsiveDialog from "@/components/shared/responsiveDialog/ResponsiveDialog"
-import SmartLink from "@/components/shared/smartLink/SmartLink"
-import { useGetMaterialQuery } from "@/lib/graphql/generated/fbi/graphql"
-import { resolveUrl } from "@/lib/helpers/helper.routes"
+import { GetMaterialQuery, useGetMaterialQuery } from "@/lib/graphql/generated/fbi/graphql"
+import { useSelectedManifestationStore } from "@/store/selectedManifestation.store"
 
-function WorkPageLayout({ wid }: { wid: string }) {
-  const [isPlayerOpen, setIsPlayerOpen] = useState(false)
+import WorkPageHeader from "./WorkPageHeader"
+import { getBestRepresentation } from "./helper"
 
-  const { data } = useQuery({
-    queryKey: useGetMaterialQuery.getKey({ wid }),
-    queryFn: useGetMaterialQuery.fetcher({ wid }),
-  })
-
-  // TODO: Handle potential error states
-  const manifestations = data?.work?.manifestations.all
-
-  const identifier = manifestations?.[0].identifiers?.[0].value || ""
+type WorkPageLayoutProps = {
+  workId: string
+  dehydratedQueryData: GetMaterialQuery | undefined
+}
 
-  const url = resolveUrl({
-    routeParams: { work: "work", ":wid": wid, read: "read" },
-    queryParams: { id: identifier },
+function WorkPageLayout({ workId, dehydratedQueryData }: WorkPageLayoutProps) {
+  const { data, isLoading } = useQuery({
+    queryKey: useGetMaterialQuery.getKey({ wid: workId }),
+    queryFn: useGetMaterialQuery.fetcher({ wid: workId }),
+    initialData: dehydratedQueryData,
   })
+  const pageContainerClasses =
+    "content-container my-grid-gap-2 flex-row flex-wrap space-y-grid-gap-2 lg:space-y-grid-gap-1 lg:my-grid-gap-half"
+  const { selectedManifestation, setSelectedManifestation } = useSelectedManifestationStore()
+
+  // Cleanup at unmount
+  useEffect(() => {
+    return () => {
+      setSelectedManifestation(null)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (!data || !data.work) return
+    // Select on the initial load
+    if (!selectedManifestation) {
+      setSelectedManifestation(getBestRepresentation(data.work))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedManifestation])
+
+  if (isLoading && !data) {
+    return <div className={pageContainerClasses}>Loading...</div>
```

Comment:
I don't think that would be possible, because we need to handle the notFound() after isLoading, and notFound can't be in the container block. 
![image](https://github.com/user-attachments/assets/bf22e990-7df4-44ec-968d-4e9b2e7e4bf1)

However, we now do have this task - https://reload.atlassian.net/browse/DDFBRA-243 about pre-fetching in JIRA which would then allow us to not render this whole component if we don't have data / data.work - that will fix what you're referring to here too.